### PR TITLE
Clean up docker resources

### PR DESCRIPTION
**Pull request by kwigley**
(partially) resolves https://github.com/fishtown-analytics/dbt-release/issues/3

### Description

- rm older dockerfiles
- add dockerfile from dbt-releases
- rename the development dockerfile to Dockerfile.dev to avoid confusion

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by kwigley** [comment]
Will merge into `dev/margaret-mead` after PR merge.

**Review by jtcohen6 — APPROVED** [review]
Nice! I think this will clear up confusion around Docker <> version support for folks just looking at the repo.

**Comment by jtcohen6** [comment]
> Will merge into `dev/margaret-mead` after PR merge.

We'll need to merge into / rebase `dev/margaret-mead` against `dev/kiyoshi-kuromiya` after final v0.19.0 release, and again with `dev/0.19.1` once we have a release candidate there. Lots of git fun ahead of us :)

**Pull request merged**
